### PR TITLE
Appeal review page

### PR DESCRIPTION
**Pull request by shanear**
connects #5169

### CODE REVIEW

File: app/models/temporary_appeal.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class TemporaryAppeal < ApplicationRecord
```
Comment (shanear) [inline app/models/temporary_appeal.rb]: We'll rename this to `Appeal` when it gets bulldozed

File: app/models/temporary_appeal.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  validates :receipt_date, :docket_type, presence: { message: "blank" }, on: :intake_review
```
Comment (shanear) [inline app/models/temporary_appeal.rb]: trying a new rails technique to enforce validations only in certain contexts. I think it's cleaner than what we've been doing and plan on migrating all the other intake models

File: app/models/temporary_appeal.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  def validate_receipt_date_within_range
+    return unless receipt_date
+
+    validate_receipt_date_not_before_ama
+    validate_receipt_date_not_in_future
+  end
+
+  def validate_receipt_date_not_before_ama
+    errors.add(:receipt_date, "before_ama") if receipt_date < HigherLevelReview::AMA_BEGIN_DATE
```
Comment (leikkisa) [inline app/models/temporary_appeal.rb]: Should we point all of them to HigherLevelReview::AMA_BEGIN_DATE? I think SC has it's own separate variable, but it makes sense for there to only be one.

File: client/app/intake/reducers/appeal.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+export const appealReducer = (state = mapDataToInitialAppeal(), action) => {
+  switch (action.type) {
+  case ACTIONS.START_NEW_INTAKE:
+    return mapDataToInitialAppeal();
+  case ACTIONS.FILE_NUMBER_SEARCH_SUCCEED:
+    return updateFromServerIntake(state, action.payload.intake);
+  default:
+  }
+
+  // The rest of the actions only should be processed if a HigherLevelReview intake is being processed
```
Comment (leikkisa) [inline client/app/intake/reducers/appeal.js]: Update this comment

**Comment by leikkisa** [comment]
I'm going to merge this so that I can have the code for working on another issue.

File: client/app/intake/reducers/appeal.js (hunk 4 above)
Comment (leikkisa) [reply client/app/intake/reducers/appeal.js]: Yeah I was thinking the same thing!

**Pull request merged**